src/mon/OSDMonitor.cc

```diff
@@ -9896,16 +9896,14 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
       }
     }
 
-    if (!updated.empty()) {
-      pending_inc.crush.clear();
-      newcrush.encode(pending_inc.crush, mon.get_quorum_con_features());
-      ss << "set osd(s) " << updated << " to class '" << device_class << "'";
-      getline(ss, rs);
-      wait_for_finished_proposal(op,
-        new Monitor::C_Command(mon,op, 0, rs, get_last_committed() + 1));
-      return true;
-    }
-
+    pending_inc.crush.clear();
+    newcrush.encode(pending_inc.crush, mon.get_quorum_con_features());
+    ss << "set osd(s) " << updated << " to class '" << device_class << "'";
```

Comment:
I may be reading the code wrong, but it feels like there's a chance this set will be empty on output, which would make it look weird?

Reply:
Yeah, this is a general problem with these mon commands.  We try to make the error output someone informative about what was done, but that can never be reliable.  We could strip them all away and make the CLI less verbose (and more accurate/precise), but I'm not sure human users would prefer that.